pgcli/packages/parseutils.py

```diff
@@ -162,7 +186,7 @@ def find_prev_keyword(sql):
 
     parsed = sqlparse.parse(sql)[0]
     for t in reversed(list(parsed.flatten())):
-        if t.is_keyword or t.value == '(':
+        if is_free_keyword(t) or t.value == '(':
             idx = parsed.token_index(t)
```

Comment:
I understand the fix now, but I'm very nervous now, here's why. 

We are iterating through the flattened tokens and when a condition is met, we look up the index of that token in the unflattened token list. Which does not sound right. 

Here's an easy and completely plausible way to break it:

`SELECT * FROM abc WHERE a,`

This is broken even before this PR. So that call to token_index() is just a ticking time-bomb. 

Tagging @darikg since his change relies on that index being available.

Is there an alternate way to get the same result?